typescript.js

```diff
@@ -2,79 +2,60 @@ module.exports = {
-        'comma-spacing': 'off',
-        '@typescript-eslint/comma-spacing': ['error'],
-        'dot-notation': 'off',
-        '@typescript-eslint/dot-notation': 'error',
-        'func-call-spacing': 'off',
-        '@typescript-eslint/func-call-spacing': 'error',
-        indent: 'off',
-        '@typescript-eslint/indent': ['error', 4],
-        'keyword-spacing': 'off',
-        '@typescript-eslint/keyword-spacing': 'error',
-        'lines-between-class-members': 'off',
-        '@typescript-eslint/lines-between-class-members': 'error',
-        'no-array-constructor': 'off',
-        '@typescript-eslint/no-array-constructor': 'error',
-        'no-dupe-class-members': 'off',
-        '@typescript-eslint/no-dupe-class-members': 'error',
-        'no-empty-function': 'off',
-        '@typescript-eslint/no-empty-function': ['error', {
-            allow: [
-                'arrowFunctions',
-                'functions',
-                'methods'
-            ]
-        }],
-        'no-extra-semi': 'off',
-        '@typescript-eslint/no-extra-semi': 'error',
-        'no-loop-func': 'off',
-        '@typescript-eslint/no-loop-func': 'error',
-        'no-redeclare': 'off',
-        '@typescript-eslint/no-redeclare': 'error',
-        'no-shadow': 'off',
-        '@typescript-eslint/no-shadow': 'error',
-        'no-unused-expressions': 'off',
-        '@typescript-eslint/no-unused-expressions': 'error',
-        'no-unused-vars': 'off',
-        '@typescript-eslint/no-unused-vars': ['error', { vars: 'all', args: 'after-used', ignoreRestSiblings: true }],
-        'no-use-before-define': 'off',
-        '@typescript-eslint/no-use-before-define': ['error', { functions: false, classes: true, variables: true }],
-        'no-useless-constructor': 'off',
-        '@typescript-eslint/no-useless-constructor': 'error',
-        quotes: 'off',
-        '@typescript-eslint/quotes': ['error', 'single', { avoidEscape: true }],
-        'no-return-await': 'off',
-        '@typescript-eslint/return-await': 'error',
-        semi: 'off',
-        '@typescript-eslint/semi': 'error',
-        'space-before-function-paren': 'off',
-        '@typescript-eslint/space-before-function-paren': ['error', {
-            anonymous: 'always',
-            named: 'never',
-            asyncArrow: 'always'
-        }],
-        'space-infix-ops': 'off',
-        '@typescript-eslint/space-infix-ops': 'error',
+
+        // None
+
+        /*
+            TypeScript rules outside of the recommended configuration
+        */
+        '@typescript-eslint/array-type': 'error',
+        '@typescript-eslint/ban-tslint-comment': 'error',
+        '@typescript-eslint/class-literal-property-style': 'error',
+        '@typescript-eslint/consistent-indexed-object-style': 'error',
+        '@typescript-eslint/consistent-type-assertions': 'error',
+        '@typescript-eslint/consistent-type-definitions': 'error',
+        '@typescript-eslint/consistent-type-imports': 'error',
```

Comment:
Consensus among Milan/Carson/Lukas/Jesse is that this repo should stay up to date with the latest TypeScript version so it's not a barrier to codebases who want to stay up to date.

It might also be helpful to add documentation that describes which rules target newer features to help repos find and disable. As a starting point, lets annotate rules that we notice requiring new TS features (like this one) in the README.md.

Reply:
Support for this rule was [only recently added](https://github.com/typescript-eslint/typescript-eslint/issues/2559) for DI languages. I validated it linting Angular with @typescript-eslint/eslint-plugin 4.15.0, the latest version.

Reply:
Type imports are for edge cases and, ["[are] something most users may never have to think about"](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html). The two cases where they are used are [modules with side-effects](https://github.com/microsoft/TypeScript/issues/9191) and [file by file transpiling](https://github.com/microsoft/TypeScript/pull/35200). These are not the common cases, and their use and this rule should be used on a case by case basis. Otherwise, declaring type imports are not helpful. In fact, it's incredibly annoying to have to correlate and update the form of import with a type's usage which can change. Additionally, it's commonly unclear which type is actually elided when imports include used and unused types from the same file.

**Proposal**

Leave this rule unconfigured.

Reply:
@rajsite 🚀